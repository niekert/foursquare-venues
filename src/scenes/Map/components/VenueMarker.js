import React from 'react';
import { venue as venuePropType } from 'customPropTypes';
import { Marker } from 'react-google-maps';

function VenueMarker({ venue: venueDetails }) {
  const { location } = venueDetails.venue;

  return <Marker position={{ lat: location.lat, lng: location.lng }} />;
}

VenueMarker.propTypes = {
  venue: venuePropType.isRequired,
};
export default VenueMarker;
