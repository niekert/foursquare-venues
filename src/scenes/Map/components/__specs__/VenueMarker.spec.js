import React from 'react';
import { shallow } from 'enzyme';
import VenueMarker from '../VenueMarker';

describe('<VenueMarker />', () => {
  let sharedProps;
  beforeEach(() => {
    sharedProps = {
      venue: {
        venue: {
          location: {
            lat: 52,
            lng: 53,
          },
        },
      },
      onClick: jest.fn(),
    };
  });

  test('Renders correctly', () => {
    const wrapper = shallow(<VenueMarker {...sharedProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders correctly when open', () => {
    const wrapper = shallow(<VenueMarker {...sharedProps} isDetailsOpen />);
    expect(wrapper).toMatchSnapshot();
  });
});
