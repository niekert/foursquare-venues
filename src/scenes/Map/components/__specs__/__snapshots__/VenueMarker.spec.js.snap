// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VenueMarker /> Renders correctly 1`] = `
<Marker
  onClick={
    Object {
      "calls": Array [],
      "name": "jest.fn()",
    }
  }
  position={
    Object {
      "lat": 52,
      "lng": 53,
    }
  }
/>
`;

exports[`<VenueMarker /> renders correctly when open 1`] = `
<Marker
  onClick={
    Object {
      "calls": Array [],
      "name": "jest.fn()",
    }
  }
  position={
    Object {
      "lat": 52,
      "lng": 53,
    }
  }
>
  <InfoBox>
    <Connect(VenueDetailsScene) />
  </InfoBox>
</Marker>
`;
